# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsIDocShell.idl
# Commit: 3bffe760e2db
# Full Hash: 3bffe760e2db5ff009f0d798cbb63b83a68de51b
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-08-15 09:46:49
# Regressor Bug: 1562292
# File Overlap Count: 1
# Description:
#   Bug 1562292: Part 1e - Use BrowsingContext as target in IsSandboxedFrom(). r=nika
#   
#   This check always has access to an in-process DocShell which is attempting to
#   perform a load, but its target may often be cross-process, and have no
#   in-process DocShell. This patch changes the target checks to use a
# ==============================================================================

diff -r ba987e21b532 -r 3bffe760e2db docshell/base/nsIDocShell.idl
--- a/docshell/base/nsIDocShell.idl	Thu Aug 01 16:22:52 2019 -0700
+++ b/docshell/base/nsIDocShell.idl	Thu Aug 01 16:36:32 2019 -0700
@@ -788,7 +788,7 @@
    * Returns true if we are sandboxed from aTargetDocShell.
    * aTargetDocShell - the browsing context we are attempting to navigate.
    */
-  [noscript,notxpcom,nostdcall] bool isSandboxedFrom(in nsIDocShell aTargetDocShell);
+  [noscript,notxpcom,nostdcall] bool isSandboxedFrom(in BrowsingContext aTargetBC);
 
   /**
    * This member variable determines whether a document has Mixed Active Content that